common/djangoapps/philu_commands/management/commands/tests/test_send_referral_follow_up_emails.py

```diff
@@ -0,0 +1,62 @@
+from __future__ import unicode_literals
+
+from datetime import timedelta
+
+import mock
+from django.core.management import call_command
+from django.test import TestCase
+from django.utils import timezone
+
+from lms.djangoapps.onboarding.tests.factories import UserFactory
+from openedx.features.smart_referral.models import SmartReferral
+from openedx.features.smart_referral.tests.factories import SmartReferralFactory
+from philu_commands.management.commands.send_referral_follow_up_email import DAYS_TO_SEND_FOLLOW_UP_EMAIL
+
+
+class SendReferralFollowUpEmails(TestCase):
+    """
+        Tests for `send_referral_follow_up_email` command.
+    """
+
+    @mock.patch('philu_commands.management.commands.send_referral_follow_up_email.task_send_referral_follow_up_email')
+    def test_send_referral_follow_up_command(self, mock_task_send_referral_follow_up_email):
+        """
+        This test will return a list of email address. The email address will follow certain criteria. We are getting
+        emails from 'SmartReferral' tables so only those entries will be picked that are at-least three days old. If an
+        email address is registered on our platform(philu) it will be ignored. Lastly if the 'is_referral_step_complete'
+        flag is 'True' that email will also be ignored.
+        """
+
+        test_email1 = 'test.referral1@example.com'
+        test_email2 = 'test.referral2@example.com'
+        test_email3 = 'test.referral3@example.com'
+        test_email4 = 'test.referral4@example.com'
+
+        today_date = timezone.now()
+        three_days_old_date = today_date.date() - timedelta(days=DAYS_TO_SEND_FOLLOW_UP_EMAIL)
+
+        # This email will be ignored as this is today's entry
+        SmartReferralFactory(contact_email=test_email1, is_referral_step_complete=False, created=today_date)
+
+        UserFactory(email=test_email2)
+        # This entry is three days old and 'is_referral_step_complete' is 'False' but user is registered on our
+        # platform so this will be ignored but flag will be updated
+        SmartReferralFactory(contact_email=test_email2, is_referral_step_complete=False,
+                             created=three_days_old_date)
+
+        # This entry is three days old and 'is_referral_step_complete' is 'True' so this will be ignored
+        SmartReferralFactory(contact_email=test_email3, is_referral_step_complete=True,
+                             created=three_days_old_date)
+
+        # This entry is three days old and 'is_referral_step_complete' is 'False' and user isn't registered on our
+        # platform so this email will be returned in list.
+        SmartReferralFactory(contact_email=test_email4, is_referral_step_complete=False,
+                             created=three_days_old_date)
```

Comment:
These scenarios test the 'three_days_old' criteria, but they do not test the 'at_least_three_days_old' criteria, meaning that you should add another scenario where the record is created let us say 5 days prior.

Reply:
@muak50